MLPerf_Results_Messaging_Guidelines.adoc

```diff
@@ -77,8 +81,8 @@ MLPerf results may not be compared against non-MLPerf results. For example, an M
 
 == When comparing MLPerf results, you must identify any submission differences
 
-When comparing results the main text, table, or figure must clearly identify any difference in version, division, category, verified or unverified status, scenario or chip count (count of the compute devices executing the largest number of ops, which could be processors or accelerators). When comparing Open and Closed division results, any ways in which the Open result would not qualify as a Closed result must be identified.
-  
+When comparing results the main text, table, or figure must clearly identify any difference in version, division, category, verified or unverified status, scenario or chip count (count of the compute devices executing the largest number of ops, which could be processors or accelerators) or submitter name. When comparing Open and Closed division results, any ways in which the Open result would not qualify as a Closed result must be identified. When making comparisons, submissions must not be portrayed as representing the performance available from a submitter unless the submission was by that submitter - e.g. a submission by SuperServers Inc that happened to use an accelerator from AccelCorp Ltd must not be portrayed as representing AccelCorp's performance.
```

Comment:
"When making comparisons, submissions must not be portrayed as representing the performance available from a submitter unless the submission was by that submitter - e.g. a submission by SuperServers Inc that happened to use an accelerator from AccelCorp Ltd must not be portrayed as representing AccelCorp's performance"

This seems to be a pretty fundamental change. For example, this means I cannot submit an Apple phone benchmark and make claims around what performance Apple provides -- unless I work for Apple?

Reply:
Could we add qualifying language here - specific "UNVERIFIED RESULTS" ?

"When making comparisons, UNVERIFIED RESULTS must not be portrayed as representing the performance available from a submitter unless the submission was by that submitter - e.g. a submission by SuperServers Inc that happened to use an accelerator from AccelCorp Ltd must not be portrayed as representing AccelCorp's performance"

Reply:
If you want to claim that it's the performance your submission achieved when you ran your software stack on Apple hardware, that's fine, but if you used a third-party stack on a 2018 iPhone, it'd be misrepresentation to claim that's Apple's performance _even if that's a verified result_.

Reply:
I think this asks the bigger question though: is MLPerf a Hardware Benchmark, or a Hardware+Software Benchmark.

For example, suppose my claim is: "We considered the following 3 software stacks for the hardware, A, B, and vendor. Our requirements prevent us from using the Vendor stack and B stack, so we had to use A stack. Therefore, comparing on software stack A, this is the performance of the Hardware X"

An example of this could be TF versus PT on TPUs. It wouldn't be fair to for Google to claim TF is the only one that counts, not everyone wants to use TF.

Reply:
I think my question is; how can we prevent someone like Google dictating which framework we need to use to be "official" because they clearly have a conflict there.

Reply:
You cannot ever make a claim about _Google's_ performance if you're not comparing with a submission from Google. If you want to make a claim about _your_ submission, which happens to run on Google hardware, that's fine, but the lede is be "DLSuperMegaOptimizer runs faster on AccelCorp than TPU v2!" rather than "AccelCorp beats Google!!!!".

Reply:
I think the language is catching me because it implies Hardware. For example,
"When making comparisons, submissions must not be portrayed as representing the performance available from a submitter unless the submission was by that submitter".

The term "performance available from a submitter" - I think there is an implied "Hardware" that needs to be inserted here somehow. Because performance is a combination of hardware and software, and this phrasing doesn't seem to account for cases where the software is made by one org and the hardware by another. Because technically there isn't a single "owner" of that performance, e.g. PT + TPUs depends on both orgs.

Reply:
That is to say - the hardware org doesn't have defacto ownership over the whole result unless they also own the software. Sometimes that's the case, sometimes it isn't.

Reply:
For example, suppose my claim isn't "TPUs are slow" but "PyTorch is slow". The language as written provides TPUs significant protection but does no seem to provide PyTorch that same protection.

Reply:
I don't think it protects hardware more than software. The intent is to e.g. protect Google from being compared against using a cherry-picked usage of TPU somewhere that they have no control over. The same applies to software: if you want to claim that your submission is faster than another submission using ONNX-RT on T4, fine, but you should not claim you've thereby beaten Microsoft or Nvidia.

Because we want MLPerf to have third party submissions on commodity stacks, but at the same time limit the potential for misleading claims based on those submissions.

I appreciate that the example is specific to hardware, and perhaps we should change it to clarify that it also includes software.

Reply:
Also - if we're adding language not scoped to "unverified results", should we change the title? (the title implies this is only unverified claims).

Reply:
Yes, that makes sense - @nv-rborkar ?

Reply:
Also - asked a question offline to clarify my understanding. I think I may be missing something in reading this.

Reply:
And thanks @DilipSequeira  and @nv-rborkar for keeping through all these questions. You guys have been very patient in working through 6 months of discussion :)